QuoteEngine.py

```diff
@@ -0,0 +1,30 @@
+import requests
+import yaml
+import random
+
+
+class QuoteEngine:
+
+    def __init__(self, logger):
+        self.quote_url = "https://cdn.rawgit.com/tulul/tulul-quotes/master/quote.yaml"  # noqa
+        # Note: rawgit does not have 100% uptime, but at
+        # least they're not throttling us.
+
+        self.logger = logger
+        self.cache = []
+        self.default_empty_error = "Nothing in cache. Forget to refresh_cache?"
+
+    def retrieve_random(self):
+        cache = self.cache
+        index = random.randrange(0, len(cache))
+        return self.format_quote(cache[index])
```

Comment:
You might want to use [`random.choice`](https://docs.python.org/3/library/random.html#random.choice) for this. It's clearer.